### ISSUE TITLE
`rotate` doesn't work as expected when `reflow: true`

### ISSUE THREAD

**Issue by Andrew15-5** [typst/typst#5363]
### Description

As I showed in https://github.com/typst/typst/pull/5330#issuecomment-2449809323, in v0.12.0 to rotate overshoots the size of the content for most degrees when `rotate(reflow: true)`. Previously it was doing the opposite: https://github.com/typst/typst/pull/5330#issuecomment-2452044452. As a result, in never worked as expected, and we need to fix that. At least for the majority of use cases.

```typ
#set rotate(reflow: true)
#set page(height: 80pt, width: 120pt, fill: gray)
#place(block(width: 100%, height: 100%, fill: green))
#align(center + horizon, rotate(59deg, image("images/tiger.jpg")))
```

This blocks https://github.com/typst/typst/pull/5330 that supposed to fix https://github.com/typst/typst/issues/5327.

I'm not sure, but `scale` and `skew` could have the same problem.

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by DUOLabs333** [comment]
Additionally, `rotate(0*1deg, reflow:true)` unexpectedly shrinks images.

**Comment by keriefie** [comment]
Yes, I am also having this issue, it appears to not `measure` correctly either, and the `measure` output is smaller than the actual size of the content when reflown.
